### ISSUE TITLE
Update bundled kube-state-metrics to v2.7.0 to mitigate deprecation warnings

### ISSUE THREAD

**Issue by jeroenj** [DataDog/datadog-agent#14487]
Starting form Kubernetes 1.23 we're getting deprecation warnings like the following:

```
warning "autoscaling/v2beta2 HorizontalPodAutoscaler is deprecated in v1.23+, unavailable in v1.26+; use autoscaling/v2 HorizontalPodAutoscaler"
```

This is because currently `kube-state-metrics` 2.4.2 is being used: https://github.com/DataDog/datadog-agent/blob/44e5b651eb85b3879f1cb6677ebf88e414972451/go.mod#L216.

The issue is that `kube-state-metrics` is still calling the deprecated `autoscaling/v2beta2` API: https://github.com/kubernetes/kube-state-metrics/issues/1711. This got fixed in https://github.com/kubernetes/kube-state-metrics/issues/1711 and was released as part of version 2.7.0: https://github.com/kubernetes/kube-state-metrics/releases/tag/v2.7.0.

**Comment by guhcampos** [comment]
I believe this can be closed, I was looking for open issues regarding bumping that to `2.8.0` as it provides the metrics I want, and during the research, I noticed the current version is `2.7.0`.

**Comment by jeroenj** [comment]
Thanks for noticing. It was indeed upgraded in https://github.com/DataDog/datadog-agent/pull/15623. It looks like this will be released as part of `7.44.0`. 👍 

Closing this issue.

**Comment by yomofo2s** [comment]
This fix the error for Kubernetes v 1.24.10-gke.2300 when the KSM is upgraded from kube-state-metrics:v2.6.0 to kube-state-metrics:v2.7.0
